engine/storage/snapshot/src/org/apache/cloudstack/storage/snapshot/XenserverSnapshotStrategy.java

```diff
@@ -268,7 +268,9 @@ public boolean deleteSnapshot(Long snapshotId) {
                 SnapshotDataStoreVO snapshotOnPrimary = snapshotStoreDao.findBySnapshot(snapshotId, DataStoreRole.Primary);
                 if (snapshotOnPrimary != null) {
                     SnapshotInfo snapshotOnPrimaryInfo = snapshotDataFactory.getSnapshot(snapshotId, DataStoreRole.Primary);
-                    if (((PrimaryDataStoreImpl)snapshotOnPrimaryInfo.getDataStore()).getPoolType() == StoragePoolType.RBD) {
+                    long volumeId = snapshotOnPrimary.getVolumeId();
+                    VolumeVO volumeVO = volumeDao.findById(volumeId);
```

Comment:
@rhtyd Hi Rohit, each snapshot should have a volume_id (at least on all our platforms).
Actually in engine/storage/snapshot/src/org/apache/cloudstack/storage/snapshot/SnapshotObject.java, the getVolumeId() also return long , not Long.